### ISSUE TITLE
Discussion about the anchor center

### ISSUE THREAD

**Issue by HC-2016** [matterport/Mask_RCNN#618]
First, thank the authors for the well documented project.
Here, I have some confusion about the anchor as the followings:

According the codes in [https://github.com/matterport/Mask_RCNN/blob/master/mrcnn/utils.py#L607](url), the anchors in all feature maps start from (0, 0).
anchor centers in all feature maps:
scale1(stride=4): (0, 0), (0, 0), (0, 0), (0, 4), ... , (0, 8), .... , (0, 16), ....
scale2(stride=8): (0, 0), (0, 0), (0, 0), (0, 8), ... , (0, 16), ....
scale3(stride=16): (0, 0), (0, 0), (0, 0), (0, 16), ....
...
It's wonderful that some anchors in different scales are aligned.

However, in another project [https://github.com/smallcorgi/Faster-RCNN_TF](url) ,
the anchors in the single scale feature start from (7.5, 7.5) which depends on the size of the reference (0, 0, 15, 15) window. See the codes In the [https://github.com/smallcorgi/Faster-RCNN_TF/blob/master/lib/rpn_msr/proposal_layer_tf.py#L36](url)
anchor centers in the feature map(stride=16):
(7.5, 7.5), ...,  (7.5, 23.5), ... , (7.5, 39.5)

In my opinion, the second definition seems more reasonable. 
Taking the perceptual field into account, I think the anchor centers should start from (stride/2, stride/2).
 
I wonder whether the performance  would change with different anchor setting?

**Comment by waleedka** [comment]
Good point! I agree that the center of the anchor would ideally be at the center of the receptive field. I don't know if it has a noticeable impact, though, because the difference is small relative to the size of the receptive field. The network probably learns to adjust for it.

If you get the chance to test it, I'd be very interested to learn if you notice any difference.
